src/ensure_buffer.ts

```diff
@@ -10,12 +9,12 @@ import { isBuffer } from './parser/utils';
  * @throws TypeError If anything other than a Buffer or Uint8Array is passed in
  */
 export function ensureBuffer(potentialBuffer: Buffer | ArrayBufferView | ArrayBuffer): Buffer {
-  if (isBuffer(potentialBuffer)) {
-    return potentialBuffer;
-  }
-
   if (ArrayBuffer.isView(potentialBuffer)) {
-    return Buffer.from(potentialBuffer.buffer);
+    return Buffer.from(
+      potentialBuffer.buffer,
+      potentialBuffer.byteOffset,
+      potentialBuffer.byteLength
+    );
```

Comment:
I know the bug was here before anyway, but I should absolutely have caught this during #429, sorry.